### PR TITLE
Add Flyway migration

### PR DESCRIPTION
**Pull request by gorshkovayulia**
@ctapobep , could you please take a look at my Flyway settings and tell me if they're correct or not? It seems like my migration cannot be detected(

### CODE REVIEW

File: src/main/java/com/example/dog/dao/JdbcDogDAO.java [hunk 1]
```diff
@@ -24,18 +24,18 @@ public class JdbcDogDAO implements DogDAO {
...
-                    "CREATE TABLE DOG (" +
-                    "ID INTEGER PRIMARY KEY AUTO_INCREMENT, " +
-                    "NAME VARCHAR (255) NOT NULL, " +
-                    "HEIGHT INTEGER NOT NULL, " +
-                    "WEIGHT INTEGER NOT NULL, " +
-                    "BIRTHDAY TIMESTAMP WITH TIME ZONE)");
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
+//        try (Connection conn = dataSource.getConnection()) {
```
Comment (ctapobep) [inline src/main/java/com/example/dog/dao/JdbcDogDAO.java]: Why not delete it?

File: src/main/resources/controllers-context.xml [hunk 2]
```diff
@@ -31,6 +31,20 @@
         <property name = "password" value = "${db.password}"/>
     </bean>
 
+    <bean id="flyway" class="org.flywaydb.core.Flyway" init-method="migrate">
+        <constructor-arg ref="flywayConfig"/>
+    </bean>
+
+    <bean id = "flywayConfig" class = "org.flywaydb.core.api.configuration.ClassicConfiguration">
```
Comment (ctapobep) [inline src/main/resources/controllers-context.xml]: You don't need to expose this bean. You'll inject it only in 1 place. So it's better to make it a nested bean to the original "flyway" bean.

File: src/test/java/com/example/dog/controller/DogControllerRestAssuredMockMvcTest.java [hunk 3]
```diff
@@ -31,10 +32,13 @@ public class DogControllerRestAssuredMockMvcTest extends AbstractTestNGSpringCon
 
     @Autowired
     private WebApplicationContext context;
+    @Autowired
+    private Flyway flyway;
 
     @BeforeClass
     public void setUp() {
         RestAssuredMockMvc.mockMvc(MockMvcBuilders.webAppContextSetup(context).build());
+        flyway.migrate();
```
Comment (ctapobep) [inline src/test/java/com/example/dog/controller/DogControllerRestAssuredMockMvcTest.java]: You already invoked `migrate()` using `init-method`

**Comment by gorshkovayulia** [comment]
The problem has been fixed, thanks

**Comment by ctapobep** [comment]
+2

**Pull request merged**